examples/async-sign/signed_message/src/lib.rs

```diff
@@ -0,0 +1,89 @@
+#![no_std]
+
+use codec::Decode;
+use gstd::{debug, exec, msg, prelude::*, ProgramId};
+use scale_info::TypeInfo;
+use sp_core::{
+    crypto::{AccountId32, UncheckedFrom},
+    sr25519,
+};
+
+#[derive(Debug, Decode, TypeInfo)]
+struct InitArgs {
+    account: AccountId32,
+}
+
+#[derive(Debug, Decode, TypeInfo)]
+pub struct HandleArgs {
+    pub message: Vec<u8>,
+    pub signature: Vec<u8>,
+}
+
+gstd::metadata! {
+    title: "SignedMessage",
+        init:
+            input: InitArgs,
+        handle:
+            input: HandleArgs,
+}
+
+static mut SIGNATORY: Option<ProgramId> = None;
+
+#[no_mangle]
+pub unsafe extern "C" fn handle() {
+    let signatory = match SIGNATORY {
+        None => {
+            msg::reply_bytes(b"Uninitialized", 10_000, 0);
```

Comment:
Not sure that `reply` is the best way to inform the user about errors, need to discuss it.